src/SpotifyBackend.cpp

```diff
@@ -0,0 +1,44 @@
+/** ----------------------------------------------------------------------------
+ * @file    SpotifyBackend.h
+ * @author  Stefan Jahn <stefan.jahn332@gmail.com>
+ * @brief   Class handles Music Playback with a Spotify Backend
+ * ---------------------------------------------------------------------------*/
+
+#include <vector>
+#include "SpotifyBackend.h"
+#include "Track.h"
+
+TResultOpt SpotifyBackend::initBackend() {
+    return std::nullopt;
+}
+
+TResult<std::vector<std::shared_ptr<Track>>> SpotifyBackend::queryTracks(std::string const &pattern, size_t const num) {
+    auto tmp = std::vector<std::shared_ptr<Track>>();
```

Comment:
storing a smart pointer in a vector adds a second level of indirection (the elements in the vector are on the heap themselves)
other than that the Tracks themselves are only stored in the `DateStore` (no shared ownership needed)

Reply:
you mean i should remove all the shared pointers?

Reply:
or would be a `unique_ptr<vector<Track>>` better?

Reply:
well, the tracks will either be directly forwarded to the client or stored in the DataStore, where they remain until removal. so, i think using just `vector<Track>` should work best

moving (not copying!) a vector is simple because you only copy its length and the data pointer, so no pointer needed here either